modules/rke2/cloud-config.tpl.yaml

```
@@ -0,0 +1,20 @@
+#cloud-config
+hostname: ${hostname}
+users:
+  - name: nutanix
+    sudo: ['ALL=(ALL) NOPASSWD:ALL']
+    shell: /bin/bash
+    ssh_authorized_keys:
+    %{ for key in authorized_keys ~}
+    - ${key}
+    %{ endfor ~}
+ssh_pwauth: False
+runcmd:
+  - |
+    # If no bootstrap IP is provided then start RKE2 as single node/bootstrap
+    if [ -z ${bootstrap_ip} ]; then
+      server_ip=$(ip route get $(ip route show 0.0.0.0/0 | grep -oP 'via \K\S+') | grep -oP 'src \K\S+')
+    else
+      server_ip=${bootstrap_ip}
+    fi
+    /root/rke2-startup.sh -t ${token} -s $${server_ip} -u nutanix ${agent}
```

```suggestion
    /root/rke2-startup.sh -t ${token} -T ${cluster_hostname} -s $${server_ip} -u nutanix ${agent}
```
Should pass in hostname(s) to be added to cluster certificate SANs.